Observer/Product/CollectionObserver.php

```diff
@@ -44,20 +47,25 @@ public function execute(Observer $observer)
         if ($this->configModel->isEnabled()) {
             $collection = $observer->getCollection();
             foreach ($collection as $product) {
+                $product = $this->productRepository->getById($product->getId());
+
                 $this->galleryReadHandler->execute($product);
+
                 $mediaGalleryImages = $product->getMediaGalleryImages();
 
-                if (!empty($mediaGalleryImages)) {
-                    foreach ($mediaGalleryImages as $mediaGalleryImage) {
-                        $file = $mediaGalleryImage->getData('path');
-                        $fileIsNotAvailable = $this->helper->fileIsNotAvailable($file);
+                foreach ($mediaGalleryImages as $mediaGalleryImage) {
+                    $file = $mediaGalleryImage->getData("path");
```

Comment:
@bansari-salecto any reason why you have removed if (!empty($mediaGalleryImages)) { ?